src/pages/instances/forms/NetworkForm.tsx

```diff
@@ -111,61 +112,66 @@ or remove the originating item"
         }),
 
         ...formik.values.devices.map((formDevice, index) => {
-          if (formDevice.type !== "nic") {
+          if (!formDevice.type?.includes("nic")) {
             return {};
           }
+          const device = formik.values.devices[index] as
+            | LxdNicDevice
+            | CustomNetworkDevice;
+
           return getConfigurationRowBase({
             override: "",
             label: (
               <Label forId={`networkDevice${index}`}>
-                <b>
-                  {isReadOnly
-                    ? (formik.values.devices[index] as LxdNicDevice).name
-                    : "Network"}
-                </b>
+                <b>{isReadOnly ? device.name : "Network"}</b>
               </Label>
             ),
-            value: isReadOnly ? (
-              (formik.values.devices[index] as LxdNicDevice).network
-            ) : (
-              <div className="network-device" key={index}>
-                <div>
-                  <Select
-                    label="Network device"
-                    name={`devices.${index}.network`}
-                    id={`networkDevice${index}`}
-                    onBlur={formik.handleBlur}
-                    onChange={formik.handleChange}
-                    value={
-                      (formik.values.devices[index] as LxdNicDevice).network
-                    }
-                    options={getNetworkOptions()}
-                  />
-                  <Input
-                    label="Network name"
-                    name={`devices.${index}.name`}
-                    id={`networkName${index}`}
-                    onBlur={formik.handleBlur}
-                    onChange={formik.handleChange}
-                    value={(formik.values.devices[index] as LxdNicDevice).name}
-                    type="text"
-                    placeholder="Enter name"
-                  />
-                </div>
-                <div>
-                  <Button
-                    className="delete-device"
-                    onClick={() => removeNetwork(index)}
-                    type="button"
-                    appearance="base"
-                    aria-label="delete network"
-                    hasIcon
-                  >
-                    <Icon name="delete" />
-                  </Button>
+            value:
+              device.type === "custom-nic" ? (
+                "custom network"
```

Comment:
Maybe we can have something more explicit than this? Even just adding an info icon next to this text, which shows a tooltip with something like "A custom network can be edited only from the YAML configuration" would make things clearer for the user?